crates/prune/prune/src/segments/user/transaction_lookup.rs

```diff
@@ -47,18 +52,26 @@ where
         // It is not possible to prune TransactionLookup data for which we don't have transaction
         // data. If the TransactionLookup checkpoint is lagging behind (which can happen e.g. when
         // pre-merge history is dropped and then later tx lookup pruning is enabled) then we can
-        // only prune from the tx checkpoint and onwards.
-        if let Some(txs_checkpoint) = provider.get_prune_checkpoint(PruneSegment::Transactions)? &&
+        // only prune from the lowest static file.
+        if let Some(lowest_range) =
+            provider.static_file_provider().get_lowest_range(StaticFileSegment::Transactions) &&
             input
                 .previous_checkpoint
-                .is_none_or(|checkpoint| checkpoint.block_number < txs_checkpoint.block_number)
+                .is_none_or(|checkpoint| checkpoint.block_number < Some(lowest_range.start()))
```

Comment:
Also, I'm pretty sure this also fixes a bug on `TransactionLookup` pruning. The code checked the `PruneSegment::Transactions` checkpoint and used it as a fallback starting point if the `TransactionLookup` checkpoint was behind.

However, the `Transactions` pruner (which we're removing) was responsible for moving transaction data from the database to static files - not for data expiration/deletion. This meant the `Transactions` checkpoint was always at or near the chain tip (since we don't actually store it in DB). Resulting in the `TransactionLookup` never being able to actually clean-up anything.